### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry): ring hom is integral if the induced map between affine lines is closed

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean [hunk 1]
```diff
@@ -636,6 +629,66 @@ lemma exists_idempotent_basicOpen_eq_of_is_clopen {s : Set (PrimeSpectrum R)}
     exact PrimeSpectrum.zeroLocus_anti_mono
       (Set.singleton_subset_iff.mpr <| Ideal.pow_le_self hnz hx)
 
+section IsIntegral
+
+open Polynomial
+
+theorem comap_singleton_isClosed_of_isIntegral {R S : Type*} [CommRing R] [CommRing S]
```
Comment (chrisflav) [inline Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:636]:
```suggestion
theorem isClosed_singleton_comap_of_isIntegral {R S : Type*} [CommRing R] [CommRing S]
```

File: Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean [hunk 2]
```diff
@@ -636,6 +629,66 @@ lemma exists_idempotent_basicOpen_eq_of_is_clopen {s : Set (PrimeSpectrum R)}
...
+  · rw [(isClosed_zeroLocus _).closure_subset_iff, Set.image_subset_iff, preimage_comap_zeroLocus]
+    exact zeroLocus_anti_mono (Set.image_preimage_subset _ _)
+  · rintro x (hx : I.comap f ≤ x.asIdeal)
+    obtain ⟨q, hq₁, hq₂⟩ := Ideal.exists_minimalPrimes_le hx
+    obtain ⟨p', hp', hp'', rfl⟩ := Ideal.exists_comap_eq_of_mem_minimalPrimes f _ hq₁
+    let p'' : PrimeSpectrum S := ⟨p', hp'⟩
+    apply isClosed_closure.stableUnderSpecialization ((le_iff_specializes
+      (comap f ⟨p', hp'⟩) x).mp hq₂) (subset_closure (by exact ⟨_, hp'', rfl⟩))
+
+lemma isIntegral_of_isClosedMap_comap_polynomial
```
Comment (chrisflav) [inline Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:657]:
```suggestion
lemma _root_.Algebra.isIntegral_of_isClosedMap_comap_polynomial
```
maybe? (to make it more discoverable)

Comment (erdOne) [reply Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:657]: I think it has comap in its name so it should live in the `PrimeSpectrum` namespace.

Comment (chrisflav) [reply Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean]: Hmm, so maybe `Algebra.isIntegral_of_isClosedMap_primeSpectrum_comap_polynomial`? Both are fine, I leave it up to you.

**Comment by chrisflav** [comment]
Thanks!

maintainer delegate

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**